src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/NestedFieldProjection.java

```diff
@@ -57,14 +62,21 @@ public NestedFieldProjection(SearchRequestBuilder request) {
     public void project(List<Field> fields) {
         if (isAnyNestedField(fields)) {
             initBoolQueryFilterIfNull();
-
+            // Extract nested query from WHERE clause
             List<NestedQueryBuilder> nestedQueries = extractNestedQueries(query());
             groupFieldNamesByPath(fields).forEach(
-                (path, fieldNames) -> buildInnerHit(fieldNames, findNestedQueryWithSamePath(nestedQueries, path))
+                (path, fieldNames) -> buildInnerHit(fieldNames, findNestedQueryWithSamePath(nestedQueries, path, fieldNames))
             );
         }
     }
 
+    /** Reset values of lastFieldInPath and lastFieldNames to prepare for next check */
+    private void setLastValues(String path, List<String> fieldNames) {
+        int pos = path.lastIndexOf(SEPARATOR);
+        lastFieldInPath = path.substring(pos + 1);
```

Comment:
Any chance the path doesn't include a separator?

Reply:
It could. But it doesn't matter, because it doesn't break the code and returns that exact path back to us.